kong/db/dao/plugins.lua

```diff
@@ -268,16 +268,16 @@ function Plugins:load_plugin_schemas(plugin_set)
         local Strategy = require(string.format("kong.db.strategies.%s", db.strategy))
         for name, schema_def in pairs(daos_schemas) do
           if name ~= "tables" and schema_def.name then
-            ngx_log(ngx_DEBUG, string.format("Loading custom plugin entity: '%s.%s'", plugin, name))
+            ngx_log(ngx_DEBUG, string.format("Loading custom plugin entity: '%s.%s'", plugin, schema_def.name))
```

Comment:
Took me a while to see how this added support for arrays (namely, that then the `name` variable becomes the numeric index returned in arbitrary order, and that it goes unused in this loop except for comparing against `"tables"`). I just wonder if keeping this around for backwards compatibility sake is worth the "sneaky" code — plus, this code assumes that pairs always traverses the numeric keys of a table in order, right? I believe this is true in practice in LuaJIT for tables specified in literals as is usually the case in daos.lua, but IIRC it is not a formal guarantee.

And AFAICT the backwards-compat will be partial anyway, since plugins with dependencies among custom daos like oauth2 does will require using ordered arrays.

BTW, thanks for the explanation in the commit message, otherwise I would have totally missed the intent!

Reply:
Removing the backward-compatibility and using `ipairs` instead of `pairs` would involve changing all the plugins' `daos.lua` files to return an array instead of a hash. That is the main reason for keeping the "sneaky" code here.

If you don't think that's a good enough reason I will change all the `daos.lua` to arrays and use `ipairs` here.